SteamKit2/Tests/AsyncJobFacts.cs

```diff
@@ -396,16 +396,27 @@ public async Task AsyncJobMultipleContinuesAsynchronously()
             var continuationThreadID = -1;
             var continuation = asyncTask.ContinueWith( t =>
             {
-                continuationThreadID = Thread.CurrentThread.ManagedThreadId;
+                continuationThreadID = Environment.CurrentManagedThreadId;
             }, TaskContinuationOptions.ExecuteSynchronously );
 
-            var completionThreadID = Thread.CurrentThread.ManagedThreadId;
+            var completionThreadID = Environment.CurrentManagedThreadId;
             asyncJob.AddResult( new Callback { JobID = 123 } );
 
-            await continuation;
+            WaitForTaskWithoutRunningInline( continuation );
 
             Assert.NotEqual( -1, continuationThreadID );
             Assert.NotEqual( completionThreadID, continuationThreadID );
         }
+
+        static void WaitForTaskWithoutRunningInline( Task task )
```

Comment:
Add `[MethodImpl(MethodImplOptions.NoInlining)]`?

Reply:
This isn't about compiler inlining but about whether the runtime runs the task "inline" in the wait function - reaching `TaskScheduler.TryRunInline`